### ISSUE TITLE
Parsed postalcode is not used in elasticsearch query

### ISSUE THREAD

**Issue by amatissart** [pelias/api#1049]
Queries composed of an address and a postalcode often lead to incorrect results, that don't take the postalcode into account (although it's parsed correctly).

With
[/v1/search?text=12 rue de dunkerque 75010](https://search.mapzen.com/v1/search/v1/search?text=12%20rue%20de%20dunkerque%2075010)
The first result has postalcode **29830** and 
`"parsed_text":{"number":"12","street":"rue de dunkerque","postalcode":"75010"}`

Whereas
[/v1/search/structured?address=12 rue de dunkerque&postalcode=75010](https://search.mapzen.com/v1/search/structured?address=12%20rue%20de%20dunkerque&postalcode=75010)
returns the expected result with postalcode **75010**.

Technically, on a local instance, the non-structured search triggers a single elasticsearch query (via `search_with_ids`) that does NOT filter on `postalcode` :

```
{
  "query": {
    "function_score": {
      "query": {
        "bool": {
          "minimum_should_match": 1,
          "should": [
            {
              "bool": {
                "_name": "fallback.address",
                "must": [
                  {
                    "match_phrase": {
                      "address_parts.number": "12"
                    }
                  },
                  {
                    "match_phrase": {
                      "address_parts.street": "rue de dunkerque"
                    }
                  }
                ],
                "should": [],
                "filter": {
                  "term": {
                    "layer": "address"
                  }
                },
                "boost": 10
              }
            },
            {
              "bool": {
                "_name": "fallback.street",
                "must": [
                  {
                    "match_phrase": {
                      "address_parts.street": "rue de dunkerque"
                    }
                  }
                ],
                "should": [],
                "filter": {
                  "term": {
                    "layer": "street"
                  }
                },
                "boost": 5
              }
            }
          ]
        }
      },
      "max_boost": 20,
      "functions": [
        {
          "field_value_factor": {
            "modifier": "log1p",
            "field": "popularity",
            "missing": 1
          },
          "weight": 1
        },
        {
          "field_value_factor": {
            "modifier": "log1p",
            "field": "population",
            "missing": 1
          },
          "weight": 2
        }
      ],
      "score_mode": "avg",
      "boost_mode": "multiply"
    }
  },
  "sort": [
    "_score"
  ],
  "size": 20,
  "track_scores": true
}
```

Is that an expected behavior ?

**Comment by orangejulius** [comment]
Hey @amatissart,
Thanks for reporting this, and sorry we missed it for a few days. I think this is a bug, and I bet we can and should fix it. @trescube is our query expert and can confirm whether or not that's true. If you're interested in submitting a pull request to fix it he's also the one to talk to, so let us know.

**Comment by amatissart** [comment]
I would need your expertise to build a proper fix.

Should `searchWithIdsShouldExecute` check for the existence of the same admin properties (city, county, etc) as `placeholderIdsLookupShouldExecute` ?
Or maybe another rule should be added so that `search_with_ids` is never executed when a postal code is detected ?

**Comment by orangejulius** [comment]
Good news! We've just fixed this with #1226 and https://github.com/pelias/query/pull/91.

**Comment by orangejulius** [comment]
![image](https://user-images.githubusercontent.com/111716/47946970-35bb9700-deea-11e8-9553-6c3f440a067b.png)
[/v1/search?text=12 rue de dunkerque 75010](http://pelias.github.io/compare/#/v1/search%3Ftext=12%20rue%20de%20dunkerque%2075010)

The first result is now the one with the `75010` postal code from Paris!

Let us know if there are other similar queries that don't work.
